mindsdb/integrations/kafka/kafkadb.py

```diff
@@ -0,0 +1,172 @@
+            del self.streams[stream_name]
+
+    def work(self):
+        self.consumer = kafka.KafkaConsumer(bootstrap_servers=f"{self.host}:{self.port}",
+                                            consumer_timeout_ms=1000)
+
+        self.consumer.subscribe([self.control_topic_name])
+        log.error(f"Integration {self.name}: subscribed  to {self.control_topic_name} kafka topic")
+        while not self.stop_event.wait(0.5):
+            try:
+                # break if no record about this integration has found in db
+                if not self.should_i_exist():
+                    break
+                self.start_stored_streams()
+                self.stop_deleted_streams()
+                try:
+                    msg_str = next(self.consumer)
+
+                    stream_params = json.loads(msg_str.value)
+                    stream = self.get_stream_from_kwargs(**stream_params)
+                    stream.start()
+                    # store created stream in database
+                    self.store_stream(stream)
+                except StopIteration:
+                    pass
+            except Exception as e:
+                log.error(f"Integration {self.name}: {e}")
+
+        # received exit event
+        self.consumer.close()
+        self.stop_streams()
+        session.close()
+        log.error(f"Integration {self.name}: exiting...")
+
+    def store_stream(self, stream):
+        """Stories a created stream."""
+        stream_name = f"{self.name}_{stream.predictor}"
+        stream_rec = Stream(name=stream_name, host=stream.host, port=stream.port,
+                            _type=stream._type, predictor=stream.predictor,
+                            integration=self.name, company_id=self.company_id,
+                            stream_in=stream.stream_in_name, stream_out=stream.stream_out_name)
+        session.add(stream_rec)
+        session.commit()
+        self.streams[stream_name] = stream.stop_event
+
+    def stop_streams(self):
+        for stream in self.streams:
+            self.streams[stream].set()
+
+    def get_stream_from_db(self, db_record):
+        kwargs = {"type": db_record._type,
+                  "predictor": db_record.predictor,
+                  "input_stream": db_record.stream_in,
+                  "output_stream": db_record.stream_out}
+        return self.get_stream_from_kwargs(**kwargs)
+
+    def get_stream_from_kwargs(self, **kwargs):
+        topic_in = kwargs.get('input_stream')
+        topic_out = kwargs.get('output_stream')
+        predictor_name = kwargs.get('predictor')
+        stream_type = kwargs.get('type', 'forecast')
+        return KafkaStream(self.host, self.port,
+                           topic_in, topic_out,
+                           predictor_name, stream_type)
+
+
+    def _query(self):
+        pass
+
+    def register_predictors(self, model_data_arr):
+        pass
+
+    def unregister_predictor(self, name):
+        pass
```

Comment:
Why are these here, why not just skip streaming integration in the respective checks? Or is it easier this way?

Reply:
Fixed